### PR TITLE
Add analytics for recent dapps

### LINKED ISSUE #1965: [analytics] Tracking segment events for dApps on home screen
### Description
Need tracking for the dApps Listed on the home screen

### User Stories 
In order to fully understand the user interaction to the dapp page, requesting the following events. 
we already have the following 
- [ ] 1. **view_all_dapps** : when user clicks on the View all dapps toggle on the homescreen
- [ ] 2. **dapp_home_impression** : when dapps are shown in home screen. Send one event for one dapp
- [ ] 3. **dapp_home_click** : when user clicks on the dapp in the bottom sheet

For the events from 2-4 please include the following parameters. 

- [ ]  categoryId: string
- [ ]   dappId: string
- [ ]   dappName: string
- [ ]   section: string ( can be home screen or recently used) 
- [ ]   horizontalPosition: number

----- Exception -------
- [ ]  **Note:** Incase of Bidali, where no alert is shown due to web app, please fire both dapp_click and dapp_open at clicking the dapp.

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

Add analytics for recently used dapps to track impressions and interactions.

### Other changes

N/A

### Tested

Manually and unit tests

### How others should test

N/A

### Related issues

- Fixes #1965 

### Backwards compatibility

Yes

### CODE REVIEW

**Review by jeanregisser — APPROVED** [review]
Great! 👍 
Already approved with some small comments.

File: packages/mobile/src/analytics/Properties.tsx [hunk 1]
```diff
@@ -1215,6 +1216,15 @@ interface DappExplorerEventsProperties {
...
 }
 
+interface RecentlyUsedDappProperties {
+  [RecentlyUsedDappEvents.dapp_view_all]: undefined
+  [RecentlyUsedDappEvents.dapp_recently_used_impression]: DappSelectEventProperties
+  [RecentlyUsedDappEvents.dapp_recently_used_select]: DappSelectEventProperties
+  [RecentlyUsedDappEvents.dapp_recently_used_open]: DappSelectEventProperties
+  [RecentlyUsedDappEvents.dapp_recently_used_bottom_sheet_dismiss]: DappSelectEventProperties
+  [RecentlyUsedDappEvents.dapp_recently_used_bottom_sheet_open]: DappSelectEventProperties
+}
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply packages/mobile/src/analytics/Properties.tsx]: thank you for pointing this out! i didn't know this pattern existed, i checked with Gowtham and he agreed your suggestion is better! so i have updated to use the same event names on both screens and have `origin` as the differentiator. it looks a lot nicer, i'm really happy you caught this!

File: packages/mobile/src/home/RecentlyUsedDapps.tsx [hunk 2]
```diff
@@ -18,15 +29,55 @@ interface Props {
...
+    )
+
+    if (numDappsVisible > lastViewedDapp.current + 1) {
+      range(lastViewedDapp.current + 1, numDappsVisible).forEach((dappIndex) => {
+        const dapp = recentlyUsedDapps[dappIndex]
+        ValoraAnalytics.track(RecentlyUsedDappEvents.dapp_recently_used_impression, {
+          categoryId: dapp.categoryId,
+          dappId: dapp.id,
+          dappName: dapp.name,
+          section: dapp.isFeatured ? 'featured' : 'all',
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply packages/mobile/src/home/RecentlyUsedDapps.tsx]: yes you're right! i have removed

File: packages/mobile/src/home/RecentlyUsedDapps.tsx [hunk 3]
```diff
@@ -18,15 +29,55 @@ interface Props {
...
+  const trackDappsImpressionForScrollPosition = debounce((horizontalContentOffset: number) => {
+    const numDappsVisible = Math.min(
+      Math.floor((windowWidth + horizontalContentOffset) / DAPP_WIDTH),
+      recentlyUsedDapps.length
+    )
+
+    if (numDappsVisible > lastViewedDapp.current + 1) {
+      range(lastViewedDapp.current + 1, numDappsVisible).forEach((dappIndex) => {
+        const dapp = recentlyUsedDapps[dappIndex]
+        ValoraAnalytics.track(RecentlyUsedDappEvents.dapp_recently_used_impression, {
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply packages/mobile/src/home/RecentlyUsedDapps.tsx]: yup i have added a comment - i actually added a comment on the `lastViewedDapp` variable but it's clearly not clear so i have updated the comment and moved it to this function :)

File: packages/mobile/src/analytics/Properties.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kathaypacific) [reply packages/mobile/src/analytics/Properties.tsx]: actually thinking more about this, the `section` event property we had is enough to differentiate the events if we allow this property to have enum value of recently used/featured/all 🙈 so we don't need the `origin` property if this `section` property is populated correctly (which i wasn't doing before, but i've done it now) - thanks for prompting me to go on this scenic simplification journey 🤩

**Comment by ValoraQA** [comment]
hey @MuckT JFYI: We are skipping this task as in "How others should test" section it is mentioned N/A.
Cc: @kathaypacific 
thanks !!

**Pull request merged**